components/calendar/src/islamic.rs

```diff
@@ -369,7 +405,10 @@ impl Calendar for IslamicUmmAlQura {
         month_code: types::MonthCode,
         day: u8,
     ) -> Result<Self::DateInner, CalendarError> {
-        let year = if era.0 == tinystr!(16, "islamic") {
+        let year = if era.0 == tinystr!(16, "islamic-umalqura")
+            || era.0 == tinystr!(16, "islamic")
```

Comment:
Thought: Maybe we shouldn't accept "islamic" here and only do the calendar specific one or "ah"

Reply:
we produce islamic for formatting, figured it would be weird if the formatting eras didn't roundtrip but that's essentially the same question as the M06L one and I think there are valid reasons to go both ways